core/src/apps/mintlayer/sign_tx/layout.py

```diff
@@ -0,0 +1,210 @@
+
+
+def format_coin_amount(
+    amount: bytes, token: MintlayerTokenOutputValue | None, coininfo: CoinInfo
+) -> str:
+    return format_coin_amount_int(int.from_bytes(amount, "big"), token, coininfo)
+
+
+def format_coin_amount_int(
+    amount_int: int, token: MintlayerTokenOutputValue | None, coininfo: CoinInfo
+) -> str:
+    if token is None:
+        decimals = coininfo.decimals
+        name = coininfo.coin_shortcut
+    else:
+        decimals = token.number_of_decimals
+        ticker = token.token_ticker.decode("utf-8")
+        name = f"Unknown token with ID: {token.token_id} and ticker {ticker}"
+
+    amount_str = format_amount(amount_int, decimals)
+
+    return f"{amount_str} {name}"
+
+
+def lock_to_string(lock: MintlayerOutputTimeLock) -> str:
+    from trezor.strings import format_timestamp
+
+    if lock.until_time is not None:
+        return f"Lock until {format_timestamp(lock.until_time)}"
+    elif lock.until_height is not None:
+        return f"Lock until block height {lock.until_height}"
+    elif lock.for_seconds is not None:
+        return f"Lock for {lock.for_seconds} seconds"
+    elif lock.for_block_count is not None:
+        return f"Lock for {lock.for_block_count} blocks"
+    else:
+        raise DataError("Unhandled lock type")
+
+
+async def confirm_output(
+    output: MintlayerTxOutput,
+    output_index: int,
+    coininfo: CoinInfo,
+    chunkify: bool,
+) -> None:
+    from ubinascii import hexlify
+
+    title = TR.bitcoin__title_confirm_details
+    if output.transfer:
+        x = output.transfer
+        assert x.address is not None
+        address_short = x.address
+        amount = format_coin_amount(x.value.amount, x.value.token, coininfo)
+        address_label = "Transfer"
+    elif output.lock_then_transfer:
+        x = output.lock_then_transfer
+        assert x.address is not None
+        address_label = "Lock then Transfer"
+        address_short = f"Destination: {x.address}\n"
+        address_short += lock_to_string(x.lock)
+        amount = format_coin_amount(x.value.amount, x.value.token, coininfo)
+    elif output.burn:
+        x = output.burn
+        address_short = "BURN"
+        amount = format_coin_amount(x.value.amount, x.value.token, coininfo)
+        address_label = ""
+    elif output.create_stake_pool:
+        x = output.create_stake_pool
+        assert x.staker is not None and x.decommission_key is not None
+        address_short = f"""Pool ID: {x.pool_id}
+Staker: {x.staker}
+Decommission key: {x.decommission_key}"
+VRF public key: {x.vrf_public_key}
+Margin ratio per thousand: {x.margin_ratio_per_thousand}
+Cost per block: {int.from_bytes(x.cost_per_block, "big")}
+"""
+        amount = format_coin_amount(x.pledge, None, coininfo)
+        address_label = "Create staking pool"
+    elif output.produce_block_from_stake:
+        x = output.produce_block_from_stake
+        address_short = f"New decommission key: {x.destination}"
```

Comment:
Actually, this is the new staker key, the decommission one can't be changed.

P.S. I guess the description here should be consistent with the one in `create_stake_pool`, which is currently "Staker". And it might be better better to change "Staker" to "Staker key", so that here it'll become "New staker key".